astropy/io/fits/header.py

```diff
@@ -1694,11 +1694,11 @@ def _keyword_from_index(self, idx):
         """
 
         if idx < 0:
-            idx += len(self._cards) - 1
+            idx += len(self._cards)
 
         keyword = self._cards[idx].keyword
         keyword = Card.normalize_keyword(keyword)
-        repeat = self._keyword_indices[keyword].index(idx)
+        repeat = len(self._keyword_indices[keyword]) - 1
```

Comment:
In my understanding, the "incorrect repeat count" mentioned in the original issue is fixed with the first line of your patch (the `idx` value). So there is no need to modify the repeat count itself.